src/components/ValueInput.vue

```diff
@@ -0,0 +1,48 @@
+<template>
+	<ItemValueLookup
+		v-if="datatype === 'wikibase-item'"
+		:value="value"
+		@input="$emit('input', $event)"
+		:error="error"
+		:disabled="disabled"
+	/>
+	<TextInput
+		v-else
+		:value="value"
+		:label="$i18n('query-builder-input-value-label')"
+		@input="$emit('input', $event)"
+		:error="error && { message: $i18n(error.message), type: error.type }"
+		:placeholder="$i18n('query-builder-input-value-placeholder')"
+		:disabled="disabled"
+	/>
```

Comment:
If value input components share the same interface (which they probably should) this whole template (or the whole component?) could turn into something like this:

```suggestion
	<component
		:is="{
			string: 'TextInput',
			'external-id': 'TextInput',
			'wikibase-item': 'ItemValueLookup'
		}[datatype]"
		:value="value"
		:label="$i18n('query-builder-input-value-label')"
		@input="$emit('input', $event)"
		:error="error && { message: $i18n(error.message), type: error.type }"
		:placeholder="$i18n('query-builder-input-value-placeholder')"
		:disabled="disabled"
	/>
```
... but ideally without doing the dispatching in the template.

If that works out, all this component does is forwarding props. At that point it seems cleaner to put the dispatching code into a regular function, and move the differences we see in the templates (error handling, messages, ...) down into the value input components. Then this can become a [functional component](https://vuejs.org/v2/guide/render-function.html#Functional-Components) or be removed entirely if nothing is left.

Reply:
I really like the looks of this, thanks! I had looked at `<component is=""` previously and it didn't seem exactly right, but your approach is very convincing.

Then the first step for this would be to wrap the `TextInput` into its own component to align the interfaces, right?. That was on my Todo-list anyway, I will look into it.

Reply:
> Then the first step for this would be to wrap the TextInput into its own component to align the interfaces, right?

Yes, that sounds good!